assgnment2/funcions.dart

```diff
@@ -0,0 +1,183 @@
+
+import 'dart:convert';
+import 'util.dart';
+import 'user.dart';         /// two classes approach (util input, util validation)
+import 'dart:io';
+bool validateUserChoice(int userChoice){
+
+ if( userChoice <1 || userChoice > 5){
+  print('please enter an integer ranging 1-5');
+  return false;
+ }
+ return true;
+
+}
+ int takeUserChoice() {
+
+ int validChoice = Util.validateInteger();// kya ye valid
+
+ while(validateUserChoice(validChoice) == false){
+   validChoice = Util.validateInteger();
+ }
+ return validChoice;
+
+}
+
+void printOptions() {
+
+  print('Enter the as per options below');
+  print('1:For Adding user details');
+  print('2:Displayng the existing user detaiils');
+  print('3:Deleting the user details');
+  print('4:Saving the user details');
+  print('5:Exit');
+  
+}
+int validRollNumber(){
+  
+  int valid= Util.validateInteger();
+
+  while(User.regieteredRollNumber.contains(valid)){
+    print('please enter a unique roll number');
+    valid = Util.validateInteger();
+  }
+   return valid; 
+}
+
+
+ List<String> readCources(){
+ Set<String> courcesOffered = {"A" , "B" ,"C" , "D","E" , "F"};  
+ Set<String> currentCources = {};
+
+ print('Cources offered are A,B,C,D,E,F');
+
+
+ while(currentCources.length < 4){
```

Comment:
This is not optimal. The user should be able to provide all his opted courses in a single input.